### PR TITLE
expression:implement vectorized evaluation for builtinInstrBinarySig

### PR DESCRIPTION
**Pull request by icditwang**
PCP #12103
### What problem does this PR solve? 
implement vectorized evaluation for builtinInstrBinarySig

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinStringFunc/builtinInstrBinarySig-VecBuiltinFunc-8         	   76005	     17651 ns/op	       0 B/op	       0 allocs/op    

BenchmarkVectorizedBuiltinStringFunc/builtinInstrBinarySig-NonVecBuiltinFunc-8      	   18814	     67997 ns/op	       0 B/op	       0 allocs/op     

BenchmarkVectorizedBuiltinStringFunc/builtinInstrBinarySig-VecBuiltinFunc#01-8      	   44893	     26276 ns/op	       0 B/op	       0 allocs/op    

BenchmarkVectorizedBuiltinStringFunc/builtinInstrBinarySig-NonVecBuiltinFunc#01-8   	   17289	     71267 ns/op	       0 B/op	       0 allocs/op  
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -1420,11 +1420,44 @@ func (b *builtinOrdSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) err
...
+			continue
+		}
+		strI := str.GetString(i)
+		substrI := substr.GetString(i)
+		idx := strings.Index(strI, substrI)
+		if idx == -1 {
+			res[i] = 0
+			continue
+		}
+		res[i] = int64(idx + 1)
```
Comment (wshwsh12) [inline expression/builtin_string_vec.go]: Use int64(utf8.RuneCountInString(str[:idx]) + 1)?

**Comment by icditwang** [comment]
@wshwsh12 I'm so sorry I delete your comment by accident.I have messaged you at the slack.

File: expression/builtin_string_vec.go [hunk 2]
```diff
@@ -1622,11 +1622,44 @@ func (b *builtinOrdSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) err
...
+	res := result.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		strI := str.GetString(i)
+		substrI := substr.GetString(i)
+		idx := strings.Index(strI, substrI)
+		if idx == -1 {
+			res[i] = 0
```
Comment (b41sh) [inline expression/builtin_string_vec.go]: This state seems like equivalent to
`res[i] = int64(idx + 1)`

Comment (icditwang) [reply expression/builtin_string_vec.go]: I will remove it.Do I need to change the non-vec func?It has it also.

Comment (b41sh) [reply expression/builtin_string_vec.go]: I think we don't need change that one

Comment (icditwang) [reply expression/builtin_string_vec.go]: done.

**Comment by b41sh** [comment]
LGTM

**Pull request merged**